Week3/server.js

```diff
@@ -0,0 +1,101 @@
+const bodyParser = require('body-parser');
+const express = require('express');
+const app = express();
+const db = require('./database.js');
+
+app.use(bodyParser.json());
+
+app.get('/user/:id', function (req, res) {
+  db.getAll(req.params.id).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error));
+});
+
+app.post('/create-user/:id', function (req, res) {
+  let id = req.params.id;
+  let firstName = req.body.firstName;
+  let lastName = req.body.lastName;
+  db.createUser(id, firstName, lastName).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error.message));
+});
+
+app.post('/create-list/:id', function (req, res) {
+  let UserId = req.params.id;
+  let category = req.body.category;
+  let prioritize = req.body.prioritize;
+  db.createList(UserId, category, prioritize).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error.message));
+});
+
+app.get('/get-lists/:id', function (req, res) {
+  db.getAll(req.params.id).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error));
+});
+
+app.post('/create-item/:id', function (req, res) {
+  let listId = req.params.id;
+  let todo = req.body.todo;
+  let description = req.body.description;
+  let prioritize = req.body.prioritize;
+  db.createItem(listId, todo, description, prioritize).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error.message));
+});
+
+app.get('/get-items/:id', function (req, res) {
```

Comment:
I got a little confused here. As I understood, the ```db.getAll``` method should expect the user id for getting the user info and items. But this endpoint ```'/get-items/:id'``` doesn't give me the complete understanding that I should send an userId for it. I would change the endpoint to this ```'/get-items/:userId'``` to avoid any confusion. =)

Reply:
And your value would be in ```req.params.userId``` for use in the function.